### DISCUSSION TITLE
regarding meaning and how to run resizing on ImageMagick

### DISCUSSION THREAD

**Discussion by MSEcharmander** [ImageMagick/ImageMagick#5014] [Help]
Hi, I ran a very large image, big tiff of 20-30 GB, that has been converted to PNG.
However, I notice there are things that I couldn't understand, which I wanted to ask here.

(1) I tried resizing the big tiff image, but at certain % it would produce resized images very quickly; for instance -resize 30% would be fast. However, when I try other resizing %, even after 5 hours, there won't be any resized image produced. It seems like it's not the memory problem because usually it would be the issue when I resize to > 100%. But without resizing (I think then it means 100% in size), the conversion to PNG would be successful, along with 30%. What is the cause and reason for the following?

(2) This is different from resizing, but I used different values of compression level, compression strategy, and compression filter. I notice that they would all produce the same number of pixels these 3 variables having similar values (but not the same) for conversion of big tiff to PNG, and the produced/converted PNG images would have different file sizes. For instance,
   (2-1) level = 1; filter =1; strategy =0 -> 172.1 MB but pixel dimension = 2208 by 36755
   (2-2) level = 0; filter= 0; strategy = 0 -> 243.6 MB but pixel dimension = 2208 by 36755
What is the reason/meaning for the following..? All I know is that PNG is lossless, so I assume the same pixel but different file sizes does not impact the bit depth/accuracy..? 

Thank you.

**Comment by urban-warrior** [comment]
ImageMagick reverts to accessing pixels on disk rather than memory when most or all of  your memory is consumed.  See https://imagemagick.org/script/architecture.php#cache for details.  For difference in PNG compression, see https://medium.com/@duhroach/how-png-works-f1174e3cc7b7.
